File: src/rule34js.ts
const toJSON = util.JSONToTable
export = function (this: void, options: { tags: string[], limit: number, pid: number }, succ: (this: void, posts: table) => void, fail?: Function) {
	HTTP({
		method: "GET",
		url: `https://rule34js.glitch.me/api?tags=${options.tags.join("+")}&limit=${options.limit || 100}&pid=${options.pid || 0}`,
		success: function (this: void, code, body, headers) {
			succ(toJSON(body))
		},
		failed: fail
	} as HTTPRequest)
}

File: lua/rule34js.lua
local toJSON
toJSON = util.JSONToTable
local function ____exports(options, succ, fail)
    HTTP(
        {
            method = "GET",
            url = (((("https://rule34js.glitch.me/api?tags=" .. tostring(
                table.concat(options.tags, "+" or ",")
            )) .. "&limit=") .. tostring(options.limit or 100)) .. "&pid=") .. tostring(options.pid or 0),
            success = function(code, body, headers)
                succ(
                    toJSON(body)
                )
            end,
            failed = fail
        }
    )
end
return ____exports
